### ISSUE TITLE
Using MorphNet for U Net Architecture

### ISSUE THREAD

**Issue by Gaurav-Naik7** [ferova/MorphNetTutorials#3]
I would like to thank you because this repository helped me to get a clear understanding of implementation of MorphNet.
I am trying to implement it on a U Net architecture which is used for semantic segmentation.

For implemanting MorphNet on classification architecture, we use tf.keras.layers.GlobalAveragePooling2D() and add a dense layer at the end such that output is (example: (None, 10) in case of cifar10 data). For calculating the MorphNet losses, this output will be assessed with one hot encoded labels of shape(10,0).

In case of semantic segmentation, my output will be a predictiction will be (for example (None, 128, 128, 2)) and the errors will be calculated with respect to ground truth of shape (None, 128, 128, 2).

Could you share some pointers from your understanding what modifications would be necessary such that it will work in this case?

Kindly write back in case you need more clarity with respect to the question.
